vendor/riscv/riscv-isa-sim/riscv/Proc.cc

```diff
@@ -61,26 +90,63 @@ namespace openhw
     {
 
         this->params.set("/top/core/0/", "isa", any(std::string("RV32GC")));
+        this->params.set("/top/core/0/", "priv", DEFAULT_PRIV);
         this->params.set("/top/core/0/", "boot_addr", any(0x80000000UL));
         this->params.set("/top/core/0/", "mmu_mode", any(std::string("sv39")));
-        this->params.set("/top/core/0/", "misa", any(""));
+
         this->params.set("/top/core/0/", "pmpaddr0", any(0x0UL));
         this->params.set("/top/core/0/", "pmpcfg0", any(0x0UL));
         this->params.set("/top/core/0/", "marchid", any(0x3UL));
         this->params.set("/top/core/0/", "mvendorid", any(0x00000602UL));
+        this->params.set("/top/core/0/", "status_fs_field_we_enable", any(false));
+        this->params.set("/top/core/0/", "status_fs_field_we", any(false));
+        this->params.set("/top/core/0/", "status_vs_field_we_enable", any(false));
+        this->params.set("/top/core/0/", "status_vs_field_we", any(false));
+
+        this->params.set("/top/core/0/", "extensions", any(std::string("")));
+
+        std::map<string,bool> registered_extensions_v;
+        registered_extensions_v["cv32a60x"] = false;
 
         // Process User Params
         ParseParams("/top/core/0/", this->params, params);
 
         string isa_str = std::any_cast<string>(this->params["/top/core/0/isa"]);
         string priv_str = std::any_cast<string>(this->params["/top/core/0/priv"]);
-        std::cout << "[SPIKE] Proc 0 | isa: " << isa_str << " priv: " << priv_str << std::endl;
+        std::cout << "[SPIKE] Proc 0 | ISA: " << isa_str << " PRIV: " << priv_str << std::endl;
         this->isa = (const isa_parser_t*) new isa_parser_t (isa_str.c_str(), priv_str.c_str());
 
         disassembler = new disassembler_t(isa);
+
         for (auto e : isa->get_extensions())
             register_extension(e.second);
 
+        string extensions_str = std::any_cast<string>(this->params["/top/core/0/extensions"]);
+        string delimiter = ",";
+        size_t found = extensions_str.rfind(delimiter);
+
+        if (found == string::npos && extensions_str != "") {
+            extensions_str = extensions_str + delimiter;
+        }
+
+        while (found != string::npos)
+        {
+            string token = extensions_str.substr(found + delimiter.length(), extensions_str.length() - 1);
+            extensions_str = extensions_str.substr(0,found);
+            auto it = registered_extensions_v.find(token);
+            if (it != registered_extensions_v.end())
+                it->second = true;
+            else
+                std::cout << "[SPIKE] Extension \"" << token << "\" can not be registered" << std::endl;
+
+            found = extensions_str.rfind(delimiter);
+        }
+
+        for (auto ext : registered_extensions_v) {
+            if (ext.second)
+                    register_extension(find_extension(ext.first.c_str())());
```

Comment:
Some custom extensions need a reset themselves (in particular `cv32a60x`).  The only point I found where the *extension* reset can be performed safely is at extension registration time.  That's how it is now done in Spike's `main()`, cf. [this line of `spike.cc`](https://github.com/openhwgroup/core-v-verif/blob/79d03a517f8f09e3ca7c24332aec91bb78b47932/vendor/riscv/riscv-isa-sim/spike_main/spike.cc#L627).

A similar trick should be applied here - unless proven unnecessary and then removed in Spike's `main` as well.

Reply:
Sure, I will take a look